[EDU-5177] feat: edge native apps architecture (#1187)
aziontech/docs: src/content/docs/en/pages/architectures/edge-native/edge-native-apps.mdx

---
title: Develop edge-native applications with Azion
description: Learn how to build your API-based applications and frontend entirely on the edge using edge-native computing.
meta_tags: 'edge, applications, web applications, api, native, CRUD'
namespace: docs_arch_edge_native
permalink: /documentation/architectures/edge-native/
---

Enhance application performance, reduce latency, tighten your security, and improve scalability using data storage and processing at the edge, closer to the end user.

Azion provides services and products to craft robust API-based applications with a frontend, hosted entirely in the edge-native computing environment. The resultant application boasts premier performance, scalable capabilities, and reliable data transactions.

---

> ## Edge Native Applications Architectures

Reviewer: ```suggestion
## Edge-Native Applications Architecture
```

Reviewer 2: meta_tags: 'edge-native, edge-native applications, applications, api, api-based applications, CRUD'

---
title: Develop edge native applications with Azion

> description: Learn how to build your API-based applications and frontend entirely on the edge by using edge native computing.

Reviewer: ```suggestion
description:Learn how to build your API-based applications and frontend entirely on the edge using edge-native computing.
```

meta_tags: 'edge, applications, web applications, api, native, CRUD'
namespace: docs_arch_edge_native
permalink: /documentation/architectures/edge-native/
---

Enhance application performance, reduce latency, tighten your security, and improve scalability using data storage and processing at the edge, closer to the end user.

> Azion provides services and products to craft robust API-based applications with a frontend, hosted entirely in the edge native computing environment. The resultant application boasts premier performance, scalable capabilities, and reliable data transactions.

Reviewer: ```suggestion
Azion provides services and products to craft robust API-based applications with a frontend, hosted entirely in the edge-native computing environment. The resultant application boasts premier performance, scalable capabilities, and reliable data transactions.
```

---

## Edge Native Applications Architectures

![Edge-native applications architecture](/assets/docs/images/uploads/edge-native-arch.png) 

### Dataflow

> 1. The user starts an application using a JavaScript template hosted on Azion's edge computing platform. The code is adjusted to trigger both Front-End and Back-End Applications within the edge system. The application interfaces directly with an API for data and command exchange.

Reviewer: ```suggestion
1. The user starts an application using a JavaScript template hosted on Azion Edge Platform. The code is adjusted to trigger both Front-End and Back-End Applications within the edge system. The application interfaces directly with an API for data and command exchange.
```

2. The API communication utilizes functions from Azion Lib, which control the communication with the database.
3. Azion Lib interacts with the Azion Edge SQL Database for operations like data retrieval, addition, update, or deletion.
4. The Azion Edge SQL Database uses parallel and synchronized data replicas for data resilience, distributing stored data.

> The edge application, via the Azion Library, ultimately returns a response containing the result of the data operations.

Reviewer: ```suggestion
The edge application, via the Azion Lib, ultimately returns a response containing the result of the data operations.
```

---

> ## Components

Reviewer: ```suggestion
## Components

```

- [Edge Application](/en/documentation/products/build/edge-application/): this component allows delivery and cache policy configuration, automatic cache policy assignment via rules based on content type, and determines content caching.

>     - [Edge Functions](/en/documentation/products/build/edge-application/edge-functions/): these are serverless, event-driven applications at the network's edge that accommodate your app's logic, including CRUD operations.

Reviewer: ```suggestion
    - [Edge Functions](/en/documentation/products/build/edge-application/edge-functions/): these are serverless, event-driven applications at the network's edge that accommodate your application's logic, including CRUD operations.
```

    - [Edge SQL](/en/documentation/products/store/edge-sql/): this serverless SQL solution, built natively for the edge, performs database operations of your CRUD application with lower latency.
- [Azion CLI](/en/documentation/products/azion-cli/overview/)
- [Azion Lib](https://github.com/aziontech/lib)
- [Azion Console](/en/documentation/products/guides/getting-to-know-azion-console/)

---

## Implementation

1. Clone the entire azion samples repository:

```bash
git clone https://github.com/aziontech/azion-samples.git
```

> 2. Navigate into the repository

Reviewer: ```suggestion
2. Navigate into the repository:
```

```bash
cd azion-samples/
```

> 3. Copy the directory you wish to another location

Reviewer: ```suggestion
3. Copy the directory you wish to another location:
```

```bash
cp -r ./samples/edge-native-with-edge-sql /path/to/your/destination
```

> 4. Run either of the following commands to install azion lib: 

Reviewer: ```suggestion
4. Run either of the following commands to install Azion Lib: 
```

```
npm install azion
```

5. Go to the location where you copied the directory to and link it to an edge application using Azion CLI:

```
azion link
```

6. Create a `.env` file to test it locally containing: 

```
AZION_TOKEN=your-api-token
AZION_DEBUG=true
```

:::note 
For the application to run properly when deployed, you must create Variables at Azion Platform with the same keys and values as presented above.
:::

7. Run it locally and deploy it: 

```
azion dev
```

```
azion deploy 
```

:::caution[warning] 

> It's important to change the database ID in the Azion lib implementation to match a databases in your account. If you don't have one created, you must create it and create a table called `people`.

Reviewer: ```suggestion
It's important to change the database ID in the Azion Lib implementation to match a database in your account. If you don't have one created, you must create it and create a table called `people`.
```

::: 

> 

Reviewer: ```suggestion
```

---
title: Develop edge-native applications with Azion
description: Learn how to build your API-based applications and frontend entirely on the edge using edge-native computing.
meta_tags: 'edge, applications, web applications, api, native, CRUD'
namespace: docs_arch_edge_native
permalink: /documentation/architectures/edge-native/
---

Enhance application performance, reduce latency, tighten your security, and improve scalability using data storage and processing at the edge, closer to the end user.

> Azion provides services and products to craft robust API-based applications with a frontend, hosted entirely in the edge-native computing environment. The resultant application boasts premier performance, scalable capabilities, and reliable data transactions.

Author: ```suggestion
Azion provides services and products to craft robust API-based applications with a front-end, hosted entirely in the edge-native computing environment. The resultant application boasts premier performance, scalable capabilities, and reliable data transactions.
```
